crates/net/common/src/metered_sender.rs

```diff
@@ -0,0 +1,64 @@
+//! Support for metering senders. Facilitates debugging by exposing metrics for number of messages
+//! sent, number of errors, etc.
+
+use metrics::Counter;
+use reth_metrics_derive::Metrics;
+use tokio::sync::mpsc::{
+    error::{SendError, TrySendError},
+    Sender,
+};
+
+/// Network throughput metrics
```

Comment:
this is not limited to network, we can use this for any bounded channel that we want to monitor.
so perhaps we move this to a separate crate?

Reply:
I moved it to `common/metrics/metered_sender.rs`, that work?